### PR TITLE
unify the internal date type 

### PR DESCRIPTION
**Pull request by jackyq2015**
Upgrade to Java 8 time API, unify the internal date type to OffsetDateTime. 

Kept java.util.Date and Calendar in some logger classes and utilities which is not related to the internal data structure.

Fix Issue #1588

### CODE REVIEW

**Comment by thadguidry** [comment]
@jackyq2015 you forgot to add Nano support to datePart()

**Comment by jackyq2015** [comment]
@thadguidry It is already in datePart():
`if ("nanos".equals(part) || "nano".equals(part) || "n".equals(part)) {`

File: main/tests/server/src/com/google/refine/tests/expr/functions/strings/ToFromConversionTests.java [hunk 1]
```diff
@@ -118,11 +117,9 @@ public void testToString() throws CalendarParserException {
...
       Assert.assertEquals(invoke("toString", Double.valueOf(100.0),"%.0f"),"100");
       
-      String expectedDate = DateFormat.getDateInstance().format(new GregorianCalendar(2013,5,1).getTime());
-      Assert.assertEquals(invoke("toString", CalendarParser.parse("2013-06-01")), expectedDate);
-      Assert.assertEquals(invoke("toString", CalendarParser.parse("2013-06-01").getTime()), expectedDate);
-      Assert.assertEquals(invoke("toString", CalendarParser.parse("2013-06-01"),"yyyy"),"2013");      
-      Assert.assertEquals(invoke("toString", CalendarParser.parse("2013-06-01"),"yyyy-MM-dd"),"2013-06-01");
+      String expectedDate = "2013-06-01T00:00:00Z";
+      Assert.assertEquals(invoke("toString", CalenderParser.parseAsOffsetDateTime("2013-06-01")), expectedDate);
+      Assert.assertEquals(invoke("toString", CalenderParser.parseAsOffsetDateTime("2013-06-01"),"yyyy-MM-dd"),expectedDate);
```
Comment (ostephens) [inline main/tests/server/src/com/google/refine/tests/expr/functions/strings/ToFromConversionTests.java]: This test is incorrect. It should be like
Assert.assertEquals(invoke("toString", CalenderParser.parseAsOffsetDateTime("2013-06-01"),"yyyy-MM-dd"),"2013-06-01");

File: main/src/com/google/refine/expr/util/CalenderParser.java [hunk 2]
```diff
@@ -491,7 +494,7 @@ void setYear(int val) throws CalendarParserException {
 /**
  * A parser for arbitrary date/time strings.
  */
-public class CalendarParser {
+public class CalenderParser {
```
Comment (wetneb) [inline main/src/com/google/refine/expr/util/CalenderParser.java]: Why was this renamed?
You probably do not want to refer to https://en.wiktionary.org/wiki/calender

Comment (jackyq2015) [reply main/src/com/google/refine/expr/util/CalenderParser.java]: It is CalendarParser now. a typo but got fixed at later commit.

Comment (wetneb) [reply main/src/com/google/refine/expr/util/CalenderParser.java]: Where is this commit? If it is in this PR, then this typo should not be visible here anymore.

Comment (ostephens) [reply main/src/com/google/refine/expr/util/CalenderParser.java]: I've made a PR on this branch to fix this #1606

**Comment by wetneb** [comment]
Let's just merge this and hope that the inevitable bugs that will arise early enough…
